ct-app/core/components/session_to_socket.py

```
@@ -0,0 +1,69 @@
+
+
+class SessionToSocket:
+
+    def __init__(self, session: Session, connect_address: str, timeout: int = 1):
+        self.session = session
+        self.connect_address = connect_address
+
+        try:
+            self.socket, self.conn = self.create_socket(timeout)
+        except (socket.error, ValueError) as e:
+            raise ValueError(f"Error while creating socket: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)